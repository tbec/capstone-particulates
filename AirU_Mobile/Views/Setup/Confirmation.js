/**
 * Final confirmation screen for setup process
 **/

import React, {Component} from 'react';
import {Text, View, TouchableHighlight, Image, AsyncStorage} from 'react-native';
import styles from '../../StyleSheets/Styles'
import { NavigationActions, StackActions } from 'react-navigation'
import { SENSOR_ARRAY } from '../../Components/Constants'

export default class Confirmation extends Component<Props> {
    constructor(props) {
        super(props)
    }

    // saves sensor after clicking final confirmation button, then navs to sensor screen
    async saveSensor() {
        // get sensor information if already saved any previously
        var sensors = []
        await AsyncStorage.getItem(SENSOR_ARRAY).then((_retrieveData) => {
            if (_retrieveData == null) {
                sensors = []
            }
            else {
                sensors = _retrieveData
            }
        })

        // get privacy, get array, JSON, save
        let privacy = this.props.navigation.getParam('privacy', 'false');
        let name = this.props.navigation.getParam('sensorName', 'NewSensor');
        let sensor = {id: 'AB-CD-EF-GF', sensorName: name};
        sensors.push(sensor);
        var json = JSON.stringify(sensors);

        // send JSON to server to add to profile

        // save sensors again to local storage
        await AsyncStorage.setItem('Sensors', json);

        // navigate back to Sensor page
        let reset = StackActions.reset({
            index: 0, 
            actions:  [NavigationActions.navigate({
                routeName: 'Tabs',
                params: {},
                action: NavigationActions.navigate('Sensor', {sensor: 'sensor'})})]
        });
        this.props.navigation.dispatch(reset);
    }

    render() {
        return (
            <View style={styles.mainView}>
                <View style={{flex: 3}}>
                    <Text>Sensor setup complete! If you have not finished mounting and securing your sensor already, please do so now.</Text>
                    <Text></Text>
                    <Text>If you need to adjust the settings for your sensor, you can do so in the settings</Text>
                    <Text/>
                    <Text>Thank you for purchasing an AirU Sensor!</Text>
                    <Text/>
                    <Image source={require('../../Resources/Confirmation.jpg')} style={{width: '100%', height: '100%', flex: 4, alignContent: 'center'}}/>
                </View>
                <View style={[styles.home, {flex: 1}]}>
                    <TouchableHighlight style={styles.button} 
                                        onPress={() => this.saveSensor()}>
                        <Text style={styles.buttonText}>Complete Setup</Text>
                    </TouchableHighlight>
                </View>
            </View>
        );
    }
}