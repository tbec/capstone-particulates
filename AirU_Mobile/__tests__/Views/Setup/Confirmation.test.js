import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Confirmation from '../../../Views/Setup/Confirmation'

// used to find testIDs in code
let findById = function(tree, testID) {
    if(tree.props && tree.props.testID === testID) {
        return tree
    }
    if(tree.children && tree.children.length > 0)
    {
        let childs = tree.children
        for(let i = 0; i < childs.length; i++)
        {
            let item = findById(childs[i], testID)
            if(typeof(item) !== 'undefined') {
                return item
            }
        }
    }
}

test('Snapshot test for Confirmation Setup', () => {
    const snap = renderer.create(<Confirmation/>).toJSON();
    expect(snap).toMatchSnapshot();
})

// COMPLETE ME
test('Save Information', () => {

})