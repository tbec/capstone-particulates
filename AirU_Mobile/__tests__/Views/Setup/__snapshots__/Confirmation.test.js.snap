// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Confirmation Setup 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "height": "80%",
      "justifyContent": "space-between",
      "paddingTop": 30,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 5,
          "flexDirection": "column",
          "padding": 40,
        },
        Object {
          "flex": 1,
          "paddingTop": 30,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../Resources/Setup_Confirmation.png",
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "140%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 7,
      }
    }
  >
    <Text>
      Sensor setup complete! If you have not finished mounting and securing your sensor already, please do so now.
    </Text>
    <Text />
    <Text>
      If you need to adjust the settings for your sensor, you can do so in the settings
    </Text>
    <Text />
    <Text>
      Thank you for purchasing an AirU Sensor!
    </Text>
    <Text />
    <Image
      source={
        Object {
          "testUri": "../../../Resources/Confirmation.jpg",
        }
      }
      style={
        Object {
          "alignContent": "center",
          "flex": 4,
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 100,
          "flexDirection": "column",
          "justifyContent": "space-evenly",
        },
        Object {
          "flex": 3,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderColor": "black",
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "space-evenly",
          "padding": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        Complete Setup
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "red",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
